### PR TITLE
Updated documentation.

### PR DESCRIPTION
**Pull request by jgebal**

### CODE REVIEW

File: docs/userguide/expectations.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Concept of expectation and matcher 
+
+Validation of the code under test (the tested logic of procedure/function etc.) is performed by comparing the actual data against the expected data.
+To do that we use concept of expectation and a matcher to perform the check on the data.
+
+It's best to give an example to get an idea what is what
+```sql
+begin
+  ut.expect( 'the tested value' ).to_( equal('the expected value') );
```
Comment (Pazus) [inline docs/userguide/expectations.md]: Why we use `to_( equal` syntax in examples and not `to_equal`?

File: docs/userguide/expectations.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+Allows regexp_like validations to be executed against the following datatypes:
+- `clob`
+- `varchar2`
+
+Usage:
+```sql
+  ut.expect( a_actual ).to_( match( a_pattern in varchar2, a_modifiers in varchar2 := null) )
+```
+
+Parameters `a_pattern` and `a_modifiers` represent a valid regexp pattern accepted by [Oracle regexp_like function](https://docs.oracle.com/cd/E11882_01/server.112/e41084/conditions007.htm#SQLRF00501)
```
Comment (Pazus) [inline docs/userguide/expectations.md]: change link to http://docs.oracle.com/database/121/SQLRF/conditions007.htm#SQLRF52141

File: docs/userguide/expectations.md (hunk 1 above)
Comment (Pazus) [inline docs/userguide/expectations.md]: I though `to_equal` is the primary syntax to use and `to_( equal` is a more general form

Comment (jgebal) [inline docs/userguide/expectations.md]: The `to_()` is a base syntax for extensible matchers and the `to_equal()` is just a wrapper.
Not all matcher have wrappers for them and getting developers familiar with the `to_` syntax seemed like a good idea.

**Pull request merged**